### ISSUE TITLE
BUG: can't install scipy on mac m1 with poetry due to incompatible constraints

### ISSUE THREAD

**Issue by artemiiiy** [scipy/scipy#19437]
### Describe your issue.

Hi guys! I tried to install scipy at my mac m1 by poetry and got error. what am I doing wrong? OR i'm not sure, but you may need to update the numpy version in your requirements to resolve this issue.

### Reproducing Code Example

```python
poetry add scipy@latest
```

### Error message

```shell
IncompatibleConstraintsError

Incompatible constraints in requirements of -root- (0.0.0):
numpy (==1.23.2) ; python_version == "3.11" and platform_python_implementation != "PyPy"
numpy (==1.22.0) ; platform_machine == "loongarch64
```

### SciPy/NumPy/Python version and system information

```shell
env:

Virtualenv
Python:         3.11.6
Implementation: CPython
Path:           /Users/artemiiiy/Library/Caches/pypoetry/virtualenvs/ndr-data-platform-backend-mVS9-62T-py3.11
Executable:     /Users/artemiiiy/Library/Caches/pypoetry/virtualenvs/ndr-data-platform-backend-mVS9-62T-py3.11/bin/python
Valid:          True

System
Platform:   darwin
OS:         posix
Python:     3.11.6
Path:       /opt/homebrew/opt/python@3.11/Frameworks/Python.framework/Versions/3.11
Executable: /opt/homebrew/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/bin/python3.11
```

**Comment by artemiiiy** [comment]
> is this related to [gh-19189](https://github.com/scipy/scipy/issues/19189)?
it's near but new

**Comment by lucascolley** [comment]
If this is a bug then maybe introduced in gh-18712?

**Comment by rgommers** [comment]
This is a real bug, unfortunately. It's very tricky to ensure that these requirements are indeed unique. And unfortunately Poetry complains even when the conflicting requirement is on some niche platform that we don't test and the user is not using.

https://github.com/scipy/oldest-supported-numpy/issues/70 was basically the same issue in `oldest-supported-numpy`.

We'll fix this by getting rid of the `==` pins for the next release. In the meantime it's probably hard to work around for Poetry users - but maybe there's a Poetry-specific workaround (I don't know).

**Comment by tylerjereddy** [comment]
To clarify, for this and gh-19189, are we planning to deal with that in `1.12.0` rather than `1.11.4`?

**Comment by rgommers** [comment]
> To clarify, for this and [gh-19189](https://github.com/scipy/scipy/issues/19189), are we planning to deal with that in `1.12.0` rather than `1.11.4`?

Probably, yes. Unless a Poetry user wants to play with it now and can propose a very minimal tweak to the constraints that is easy to verify.

**Comment by rgommers** [comment]
Fixed by gh-19550, will land in `1.11.4`.
